src/odemis/driver/smaract.py

```diff
@@ -1590,9 +1601,42 @@ def _doMoveAbs(self, future, pos):
         max_dur = dur * 2 + 1
         logging.debug("Expecting a move of %f s, will wait up to %g s", dur, max_dur)
 
+<<<<<<< HEAD
         try:
             with future._moving_lock:
                 if future._must_stop:
+=======
+        with future._moving_lock:
+            try:
+                self.Move(pos, hold_time=float("inf"))
+                while not future._must_stop.is_set():
+                    ev = self.WaitForEvent(timeout)
+
+                    # check if move is done
+                    if ev.type == MC_5DOF_DLL.SA_MC_EVENT_MOVEMENT_FINISHED:
+                        break
+
+                    now = time.time()
+                    if now > timeout:
+                        logging.warning("Stopping move due to timeout after %g s.", max_dur)
+                        self.stop()
+                        raise TimeoutError("Move is not over after %g s, while "
+                                           "expected it takes only %g s" %
+                                           (max_dur, dur))
+
+                    # Update the position from time to time (10 Hz)
+                    if now - last_upd > 0.1:
+                        self._updatePosition()
+                        last_upd = time.time()
+
+                    # Wait half of the time left (maximum 0.1 s)
+                    left = end - time.time()
+                    sleept = max(0.001, min(left / 2, 0.1))
```

Comment:
Maybe I miss something but to me the naming sleep_time or just sleep would be more logical than sleept